docs/datadog_generic_resource_dev.md

```diff
@@ -1,11 +1,50 @@
-TODO: depending on refactor https://github.com/DataDog/datadog-operator/pull/1640
-1. Add in a new const `SupportedResourcesType` in `api/datadoghq/v1alpha1/datadoggenericresource_types.go`
-2. Add it to validation enum `allowedCustomResourcesEnumMap` (`api/datadoghq/v1alpha1/datadoggenericresource_validation.go`)
-3. Add a new client within `pkg/datadogclient/client.go/InitDatadogGenericClient`
-4. Add this new client to the `internal/controller/datadoggenericresource/controller.go` file inside `Reconciler` struct
-5. Add a new `internal/controller/datadoggenericresource/resource_replace_me.go`: 
-	* `ResourceReplaceMeHandler`
-	* CRUD operations
-6. Add new resource inside `internal/controller/datadoggenericresource/utils.go` to `getHandler`
-
-Example PR: https://github.com/DataDog/datadog-operator/pull/1641
\ No newline at end of file
+# How to contribute a new DatadogGenericResource
+
+This document provides the steps to develop a new type for the `DatadogGenericResource` controller. It is recommended to go over [How to contribute][1] and [Overview of the DatadogGenericResource controller][2] to start with.
+
+## Overview
+
+The general steps to add a new type can be summarized to:
+1. Add the new resource type to the CRD.
+2. Add the related API client to the GenericResource client.
+3. Add the CRUD operations for the new resource type within the GenericResource controller and reference them.
+
+Example pull request: https://github.com/DataDog/datadog-operator/pull/1635.
+
+## Adding the new resource type
+
+1. The CRD is defined inside `api/datadoghq/v1alpha1/datadoggenericresource_types.go`: add a new constant of type `SupportedResourcesType` and reference its value in the `kubebuilder:validation:Enum` marker inside the `DatadogGenericResourceSpec` struct.
+2. Add this constant to the validation map `allowedCustomResourcesEnumMap` (`api/datadoghq/v1alpha1/datadoggenericresource_validation.go`).
+3. Generate the updated CRD with `make generate`.
+
+## Adding the API client to `DatadogGenericClient`
+1. Add the new client (from the [Datadog Go client](https://github.com/DataDog/datadog-api-client-go)) to the `DatadogGenericClient` struct inside `pkg/datadogclient/client.go`.
+2. Reference it in `InitDatadogGenericClient` function within the same file.
+
+## Adding the resource-CRUD operations and referencing them
+
+1. `internal/controller/datadoggenericresource/controller.go`:
+    * Add the new API client to the `Reconciler` struct.
+	* Reference it in the `NewReconciler` function.
+2. Create a new `internal/controller/datadoggenericresource/<your_resource>.go`file:
+    * Define an empty `<Resource>Handler` struct for the `ResourceHandler` interface.
+	* Define the 4 operations functions:
```

Comment:
```suggestion
    * Define an empty `<Resource>Handler` struct for the `ResourceHandler` interface.
	* Define the 4 operations functions:
```
I think these 2 should be in an ordered list, cos they're instructions. There's a big block of unordered lists below, so this just makes it a little easier to read

Reply:
Fixed in 9e20fc44a3f31e11f605ec0dbb981fe3e02cbefa